core/config/project_settings.cpp

```diff
@@ -153,8 +153,27 @@ const PackedStringArray ProjectSettings::_trim_to_supported_features(const Packe
 #endif // TOOLS_ENABLED
 
 String ProjectSettings::localize_path(const String &p_path) const {
-	if (resource_path.is_empty() || p_path.begins_with("res://") || p_path.begins_with("user://") ||
-			(p_path.is_absolute_path() && !p_path.begins_with(resource_path))) {
+	if (resource_path.is_empty() || (p_path.is_absolute_path() && !p_path.begins_with(resource_path))) {
+		return p_path.simplify_path();
+	}
+
+	// Check if we have a special path (like res://) or a protocol identifier.
+	String drive;
+	int p = p_path.find("://");
+	bool found = false;
```

Comment:
```suggestion
	bool found = false;
```
I think we could remove `only_chars`, make this `true` by default and set it to `false` and break if a non-ascii char is found? Not that saving a single bool really matters... :)